docs/proposals/004-repos-in-any-namespace.md

```diff
@@ -0,0 +1,206 @@
+---
+title: Neat-enhancement-idea
+authors:
+  - "@patrickbardo"
+sponsors:
+  - TBD       
+reviewers:
+  - TBD
+approvers:
+  - TBD
+
+creation-date: 2023-03-15
+last-updated: 2023-03-15
+---
+# Declarative Repository Connections in Any Namespace
+
+Extend the ArgoCD multi-tenancy model to allow ArgoCD to connect to repositories based on secrets created in
+namespaces other than the control plane's namespace.
+
+## Open Questions
+
+How to handle duplicate secrets in multiple namespaces. I would assume dealing with it in the same way applications was handle, which was prepending the namespace
+to the name of the secret, however I have not considered all aspects of this.
+
+Another open question is how to deal with Repo Secrets created in the UI after these changes. Currently,
+ArgoCD will just create the resource in the ArgoCD namespace and (I believe) default project. I believe it could
+be beneficial to limit this, but not make any breaking changes.
```

Comment:
I would say this particular challenge can be postponed; if we can manage secrets declarative, it's fine for the first step of this improvement.